src/analysis/souffle/sensitivity_analysis.dl

```diff
@@ -112,13 +95,29 @@ sqlValueSensitivity(value, sensitivity_score) :-
       sqlValueHasSensitivityScoreAtIndex(value, input_sensitivity, input_list_length-1), 
       sensitivity_score = input_sensitivity * function_sensitvity.
 
+.decl isGlobalEpsilon(epsilon: number)
+//TODO: is there a way to set a deafault option?
+isGlobalEpsilon(epsilon) :- isOperation([_, "global_epsilon", _, _, [["epsilon", $NumberAttributePayload(epsilon)], _]]).
+
+.decl isDiffPrivacyMechanism(value: AccessPath, epsilon: number)
+isDiffPrivacyMechanism(value, epsilon) :- isOperation([_, "privacy_mechanism", value, _, [["epsilon", $NumberAttributePayload(epsilon)], _]]). 
+
+//sequential composition
+.decl epsilonDiffPrivacy(value: AccessPath, epsilon: number)
+epsilonDiffPrivacy(value, epsilon) :- isDiffPrivacyMechanism(value, _),
+      epsilon = sum y : { isDiffPrivacyMechanism(value, y)}. 
+
+violatesPolicy(value) :- isSqlOperator([_, _, value, _, _]), 
+      !sqlValueSensitivity(value, _),
```

Comment:
What do the disjunctions in this disjunction correspond to?

Reply:
1. There is a privacy mechanism applied to value, but the value does not need sensitivity analysis
2. There is a privacy mechanism applied to value and the value has a particular sensitivity score 
3. (Just realized this one is incomplete, will fix on the next commit) There is no privacy mechanism on a value that has a particular sensitivity score.